Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -1370,3 +1381,36 @@ extension ApolloCodegenConfiguration.OutputOptions {
     }
   }
 }
+
+private struct AnyCodingKey: CodingKey {
+    var stringValue: String
+
+    init?(stringValue: String) {
+        self.stringValue = stringValue
+    }
+
+    var intValue: Int?
+
+    init?(intValue: Int) {
+        self.intValue = intValue
+        self.stringValue = "\(intValue)"
+    }
+}
+
+private func throwIfContainsUnexpectedKey<T, C: CodingKey & CaseIterable>(
+  container: KeyedDecodingContainer<C>,
+  type: T.Type,
+  decoder: Decoder
+) throws {
+  // Map all keys from the input object
+  let allKeys = Set(try decoder.container(keyedBy: AnyCodingKey.self).allKeys.map(\.stringValue))
+  // Map all valid keys from the given `CodingKey` enum
+  let validKeys = Set(C.allCases.map(\.stringValue))
+  guard allKeys.isSubset(of: validKeys) else {
+    throw DecodingError.typeMismatch(type, DecodingError.Context.init(
+      codingPath: container.codingPath,
+      debugDescription: "Unrecognized key found",
```

Comment:
I'm wondering if this could be put into a protocol that automatically calls this function so we don't have to remember to call it in each of the decoder functions? We could call it `StrictlyKeyedDecodable`.

I'm not sure if that would cause more complexity with the `CodingKeys`?

Reply:
> I'm wondering if this could be put into a protocol that automatically calls this function so we don't have to remember to call it in each of the decoder functions?

That was my first thought, but I shied away from it when I realized that `Codable` (`Encodable` + `Decodable`) doesn't have any reference to `CodingKeys`. Adds a lot of complexity for a function that we'd otherwise call a handful of times in one file.

Reply:
Yupp, that makes sense. If we were using this all over might be worth it, but for this one place, totally fine.